plugins/pip/venvShellHook.sh

```diff
@@ -1,8 +1,92 @@
-#!/bin/sh
+set -eu
+STATE_FILE="$DEVBOX_PROJECT_ROOT/.devbox/venv_check_completed"
+echo $STATE_FILE
 
-if ! [ -d "$VENV_DIR" ]; then
-    echo "Creating new venv environment in path: '${VENV_DIR}'"
-    python3 -m venv "$VENV_DIR"
-    echo "You can activate the virtual environment by running '. \$VENV_DIR/bin/activate' (for fish shell, replace '.' with 'source')" >&2
+is_valid_venv() {
+    [ -f "$1/bin/activate" ] && [ -f "$1/bin/python" ]
+}
+
+# Function to check if Python is a symlink to a Devbox Python
+is_devbox_python() {
+    if [ -z "$DEVBOX_PACKAGES_DIR" ]; then
+        echo "DEVBOX_PACKAGES_DIR is not set. Unable to check for Devbox Python."
+        return 1
+    fi
+    local python_path="$1/bin/python"
+    local link_target
+
+    while true; do
+        if [ ! -L "$python_path" ]; then
+            # Not a symlink, we're done
+            break
+        fi
+
+        link_target=$(readlink "$python_path")
+        echo "Checking symlink: $link_target"
+
+        if [[ "$link_target" == /* ]]; then
+            # Absolute path, we're done
+            python_path="$link_target"
+            break
+        elif [[ "$link_target" == python* ]] || [[ "$link_target" == ./* ]] || [[ "$link_target" == ../* ]]; then
+            # Relative path or python symlink, continue resolving
+            python_path=$(dirname "$python_path")/"$link_target"
+        else
+            # Unexpected format, stop here
+            break
+        fi
+    done
+
+    [[ $python_path == $DEVBOX_PACKAGES_DIR/* ]]
+}
+
+# Function to check Python version
+check_python_version() {
+    python_version=$(python -c 'import sys; print(".".join(map(str, sys.version_info[:2])))')
+    if [ "$(printf '%s\n' "3.3" "$python_version" | sort -V | head -n1)" = "3.3" ]; then
+        return 0
+    else
+        return 1
+    fi
+}
+
+# Check if we've already run this script
+if [ -f "$STATE_FILE" ]; then
+    # "We've already run this script. Exiting..."
+    exit 0
 fi
 
+# Check Python version
+if ! check_python_version; then
```

Comment:
Are we just checking the version to see if venv is available? If so, it might be easier to just try importing it.

```suggestion
if ! python -c "import venv" &>/dev/null; then
```

Reply:
Ah that's a much nicer approach

Reply:
Had to tweak this because for some reason it was failing the automated tests (but not when run locally)